blockchain-importer/src/Pos/BlockchainImporter/Tables/TxsTable.hs

```diff
@@ -11,47 +11,77 @@ module Pos.BlockchainImporter.Tables.TxsTable
+type TxRowPGR = TxRowPoly (Column PGText)
+                          (Column (PGArray PGText))
+                          (Column (PGArray PGInt8))
+                          (Column (PGArray PGText))
+                          (Column (PGArray PGInt8))
+                          (Column (Nullable PGInt8))
+                          (Column (Nullable PGTimestamptz))
 
 $(makeAdaptorAndInstance "pTxs" ''TxRowPoly)
 
 txsTable :: Table TxRowPGW TxRowPGR
-txsTable = Table "txs" (pTxs TxRow { trHash     = required "hash"
-                                   , trBlockNum = required "block_num"
-                                   , trTime     = required "time"
-                                   })
+txsTable = Table "txs" (pTxs TxRow  { trHash            = required "hash"
+                                    , trInputsAddr      = required "inputs_address"
+                                    , trInputsAmount    = required "inputs_amount"
+                                    , trOutputsAddr     = required "outputs_address"
+                                    , trOutputsAmount   = required "outputs_amount"
+                                    , trBlockNum        = required "block_num"
+                                    , trTime            = required "time"
+                                    })
 
 -- | Inserts a given Tx into the Tx history tables.
 insertTx :: PGS.Connection -> Tx -> TxExtra -> Word64 -> IO ()
 insertTx conn tx txExtra blockHeight = PGS.withTransaction conn $ do
-  insertTxHeader conn tx txExtra blockHeight
-  TDT.insertTxDetails conn tx txExtra
+  insertTxToHistory conn tx txExtra blockHeight
+  TAT.insertTxAddresses conn tx txExtra
 
 -- | Inserts the basic info of a given Tx into the master Tx history table.
-insertTxHeader :: PGS.Connection -> Tx -> TxExtra -> Word64 -> IO ()
-insertTxHeader conn tx txExtra blockHeight = void $ runInsertMany conn txsTable [row]
+insertTxToHistory :: PGS.Connection -> Tx -> TxExtra -> Word64 -> IO ()
+insertTxToHistory conn tx txExtra blockHeight = void $ runInsertMany conn txsTable [row]
   where
-    row = TxRow { trHash     = pgString $ hashToString (hash tx)
-                , trBlockNum = toNullable $ pgInt8 $ fromIntegral blockHeight
-                , trTime     = maybeToNullable utcTime
+    inputs  = toaOut <$> (catMaybes $ NE.toList $ teInputOutputs txExtra)
+    outputs = NE.toList $ _txOutputs tx
+    row = TxRow { trHash          = pgString $ hashToString (hash tx)
+                , trInputsAddr    = pgArray (pgString . addressToString . txOutAddress) inputs
+                , trInputsAmount  = pgArray (pgInt8 . coinToInt64 . txOutValue) inputs
+                , trOutputsAddr   = pgArray (pgString . addressToString . txOutAddress) outputs
+                , trOutputsAmount = pgArray (pgInt8 . coinToInt64 . txOutValue) outputs
+                , trBlockNum      = toNullable $ pgInt8 $ fromIntegral blockHeight
+                , trTime          = maybeToNullable utcTime
```

Comment:
Can this two values be empty? if so, are we going to store NULL in DB?

Reply:
Removed nullable option from block number as it will never be empty, added FIXME for the case the tx timestamp is (it should never be `None` at this stage)